src/main/java/com/mixpanel/android/mpmetrics/MixpanelNotificationRouteActivity.java

```diff
@@ -0,0 +1,163 @@
+import com.mixpanel.android.mpmetrics.MixpanelPushNotification.PushTapTarget;
+
+import org.json.JSONException;
+import org.json.JSONObject;
+
+public class MixpanelNotificationRouteActivity extends Activity {
+
+    protected final String LOGTAG = "MixpanelAPI.MixpanelNotificationRouteActivity";
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+
+        final Intent routeIntent = getIntent();
+        Bundle extras = routeIntent.getExtras();
+
+        if (null == routeIntent) {
+            MPLog.d(LOGTAG, "Notification route activity given null intent.");
+            return;
+        }
+
+        trackAction(routeIntent);
+
+        final Intent notificationIntent = handleRouteIntent(routeIntent);
+
+        if (!extras.getBoolean("sticky")) {
+            cancelNotification(extras);
+        }
+        startActivity(notificationIntent);
+    }
+
+    protected Intent handleRouteIntent(Intent routeIntent) {
+        CharSequence actionTypeChars = routeIntent.getExtras().getCharSequence("actionType");
+        PushTapTarget target;
+        if (null == actionTypeChars) {
+            MPLog.d(LOGTAG, "Notification action click logged with no action type");
+            target = PushTapTarget.HOMESCREEN;
+        } else {
+            target = PushTapTarget.fromString(actionTypeChars.toString());
+        }
+
+        CharSequence uri = routeIntent.getExtras().getCharSequence("uri");
+
+        final Intent defaultIntent = this.getPackageManager().getLaunchIntentForPackage(this.getPackageName());
+
+        switch (target) {
+            case HOMESCREEN:
+                return defaultIntent;
+            case URL_IN_BROWSER:
+                if (isValidURL(uri.toString(), PushTapTarget.URL_IN_BROWSER)) {
+                    return new Intent(Intent.ACTION_VIEW, Uri.parse(uri.toString()));
+                } else {
+                    MPLog.d(LOGTAG, "Wanted to open url in browser but url is invalid: " + uri.toString() + ". Starting default intent");
+                    return defaultIntent;
+                }
+            case URL_IN_WEBVIEW:
+                if (isValidURL(uri.toString(), PushTapTarget.URL_IN_WEBVIEW)) {
+                    return new Intent(this.getApplicationContext(), MixpanelWebViewActivity.class).
+                            putExtra("uri", uri.toString()).
+                            addFlags(Intent.FLAG_ACTIVITY_NEW_TASK);
+                } else {
+                    MPLog.d(LOGTAG, "Wanted to open url in webview but url is invalid or not secure (http): " + uri.toString() + ". Starting default intent");
+                    return defaultIntent;
+                }
+            case DEEP_LINK:
+                return new Intent(Intent.ACTION_VIEW, Uri.parse(uri.toString()));
+            default:
+                return defaultIntent;
+        }
+    }
+
+    protected boolean isValidURL(CharSequence url, PushTapTarget target) {
```

Comment:
the better api for this is probably `isValidURL(CharSequence url, boolean requireHttps)` instead of tightly coupling the PushTapTarget stuff